tests/devapps/MSI/HelperService/MSIHelperService/Helper/MSIHelper.cs

```diff
@@ -0,0 +1,370 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Identity.Client;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Net;
+using System.Net.Http.Headers;
+using System.Net.Http.Json;
+using System.Net.WebSockets;
+using System.Text;
+using System.Text.Json.Serialization;
+using System.Web;
+
+namespace MSIHelperService.Helper
+{
+    internal static class MSIHelper
+    {
+        private static HttpClient s_httpClient = new HttpClient();
+
+        //content type
+        internal const string ContentTypeJson = "application/json";
+        internal const string ContentTypeTextOrHtml = "text/html";
+        internal const string ContentTypeMulipartOrFormData = "multipart/form-data";
+        internal const string DefaultAzureResource = "webapp";
+        internal const string ManagedIdentityAuthenticationHeader = "X-IDENTITY-HEADER";
+
+        //Environment variables
+        internal static string? s_requestAppID = Environment.GetEnvironmentVariable("requestAppID");
+        internal static string? s_requestAppSecret = Environment.GetEnvironmentVariable("requestAppSecret");
+        internal static string? s_functionAppUri = Environment.GetEnvironmentVariable("functionAppUri");
+        internal static string? s_functionAppEnvCode = Environment.GetEnvironmentVariable("functionAppEnvCode");
+        internal static string? s_functionAppMSICode = Environment.GetEnvironmentVariable("functionAppMSICode");
+        internal static string? s_webhookLocation = Environment.GetEnvironmentVariable("webhookLocation");
+        internal static string? s_oMSAdminClientID = Environment.GetEnvironmentVariable("OMSAdminClientID");
+        internal static string? s_oMSAdminClientSecret = Environment.GetEnvironmentVariable("OMSAdminClientSecret");
+
+        //Protected from this API so the getEnvironmentVariable does not return these
+        internal static string[] s_protectedEnvironmentVariables = new string[] {
+            "requestAppID",
```

Comment:
Security wise, it's better to have an allow-list rather than a deny list, but this would make the test less valuable.

Reply:
I think we can have all the environment variables expected by managed identity flow in the allow list.